lib/glific/templates/interactive_templates.ex

```diff
@@ -35,6 +35,16 @@ defmodule Glific.Templates.InteractiveTemplates do
     query = Repo.filter_with(query, filter)
     # these filters are specific to interactive templates only.
     Enum.reduce(filter, query, fn
+      {:term, term}, query ->
+        from(q in query,
+          where:
+            ilike(field(q, :label), ^"%#{term}%") or
+              fragment(
+                "exists(select * from jsonb_each_text(interactive_content) where value ILIKE ?)",
```

Comment:
@AkhileshNegi can check if there is a better query to directly compare values something like

orwhere: fragment("interactive_content->>'body' like "#{term}")

OR

In case you want to search on all the values then 

orwhere: fragment("interactive_content::text like "#{term}")

Current query looks bit expensive to me.

Reply:
You can read more about here https://alchemist.camp/articles/querying-nested-map-ecto

Reply:
@pankaj-ag Yeah I checked both these alternatives and the probled is how interactive content is structured in interactive templates table. It is different for both `interactive list` and `interactive button`. Also in case of list based on number of list items we need to check same.

Reply:
@AkhileshNegi may be I am missing something but how the current query makes difference then 
**or_where: fragment("interactive_content::text like ?", ^"%#{term}%")**

the only thing that this query is taking bit more time.

Reply:
@pankaj-ag this will work fine for interactive message when type is button but for list the interactive content looks like this
<img width="602" alt="image" src="https://user-images.githubusercontent.com/40158831/229449702-8e81d3f5-2014-4072-99ae-cae16447bb68.png">
also with each list item adds a new subkey

Reply:
But we are casting the complete map to text then how it's not working ? @AkhileshNegi

Reply:
I was using this way 
```
SELECT * FROM interactive_templates
WHERE interactive_content-> 'content' ->> 'text'  LIKE '%body text%';
```